Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -18,7 +18,45 @@ open LinearMap (BilinMap)
 
 namespace QuadraticMap
 
-variable {ι R M N} [LinearOrder ι]
+section
+
+variable {ι R M N}
+
+variable [CommRing R] [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N] [DecidableEq ι]
+
+-- c.f. `_root_.map_finsupp_sum`
+open Finsupp in
+theorem map_finsuppSum (Q : QuadraticMap R M N) (f : ι →₀ R) (g : ι → R → M) :
+    Q (f.sum g) =
+      f.sum (fun i r => Q (g i r)) +
+      ∑ p ∈ f.support.sym2 with ¬ p.IsDiag,
+        p.lift ⟨fun i j => polar Q (g i (f i)) (g j (f j)), fun _ _ => polar_comm _ _ _⟩ :=
+  QuadraticMap.map_sum _ _ _
+
+-- c.f. `Finsupp.apply_linearCombination`
+open Finsupp in
+theorem apply_linearCombination (Q : QuadraticMap R M N) {g : ι → M} (l : ι →₀ R) :
+    Q (linearCombination R g l) =
+      (l.sum fun i r => (r * r) • Q (g i)) +
+      ∑ p ∈ l.support.sym2 with ¬ p.IsDiag,
+        p.lift
+          ⟨fun i j => (l i * l j) • polar Q (g i) (g j), fun i j => by
+            simp only [polar_comm, mul_comm]⟩ := by
+  simp_rw [linearCombination_apply, map_finsuppSum, polar_smul_left, polar_smul_right, map_smul,
+    mul_smul]
```

Comment:
I think this lemma would be much better if the RHS could somehow be expressed as a (pair of) linear combinations as well. I believe the first term is `linearCombination (l * l) (fun i => Q (g i))`.

Reply:
`linearCombination R (Q ∘ g) (l * l)` - but the proof is then a little trickier.